erpnext/compliance/doctype/compliance_info/compliance_info.py

```diff
@@ -59,32 +59,53 @@ def get_existing_licensees(license_number, party_type):
 
 def validate_license_expiry(doc):
 	if doc.doctype in ("Sales Order", "Sales Invoice", "Delivery Note"):
-		validate_entity_license("Customer", doc.customer)
+		get_entity_license(doc, "Customer", doc.customer)
 	elif doc.doctype in ("Supplier Quotation", "Purchase Order", "Purchase Invoice", "Purchase Receipt"):
-		validate_entity_license("Supplier", doc.supplier)
+		get_entity_license(doc, "Supplier", doc.supplier)
 	elif doc.doctype == "Quotation" and doc.quotation_to == "Customer":
-		validate_entity_license("Customer", doc.party_name)
+		get_entity_license(doc, "Customer", doc.party_name)
 
+@frappe.whitelist()
+def get_entity_license(doc, party_type, party_name):
+	# get the default license for the given party
+	license_record = validate_entity_license(party_type, party_name)
+
+	# show a warning if a license exists and is expired, and only if compliance items are present
+	is_compliance = validate_doc_compliance(doc)
+	if is_compliance:
+		license_expiry_date, license_number = frappe.db.get_value(
+		"Compliance Info", license_record, ["license_expiry_date", "license_number"])
+
+		if not license_expiry_date:
+			frappe.msgprint(_("We could not verify the status of license number {0}, Proceed with Caution.").format(
+				frappe.bold(license_number)))
+		elif license_expiry_date < getdate(nowdate()):
+			frappe.msgprint(_("Our records indicate {0}'s license number {1} has expired on {2}, Proceed with Caution.").format(
+				frappe.bold(party_name), frappe.bold(license_number), frappe.bold(license_expiry_date)))
+
+		return license_record
```

Comment:
De-indent this, so it'll atleast return the license to set, even if there's no compliance items